Update documentation to reflect recent harmonization-related changes (#172)
AlexsLemonade/refinebio-docs: docs/main_text.md

### refine.bio-harmonized Metadata

_The documentation in this section reflects data that has been processed via refine.bio as of version TODO: add version._
_See the documentation sidebar for the current version of refine.bio._

> _The `refinebio_processor_version` field in the metadata captures the refine.bio version when a sample was processed._

Reviewer: ```suggestion
_The `refinebio_processor_version` field in the downloaded metadata file captures the refine.bio version when a sample was processed._
```

Pretty sure we don't show this field anywhere on the UI

Scientists who upload results don't always use the same names for related values.
This makes it challenging to search across datasets.
We have put some processes in place to smooth out some of these issues.

![harmonized-metadata](https://user-images.githubusercontent.com/15315514/44549202-5eefc800-a6ee-11e8-8a7b-57826f0153f2.png)

> We combine certain fields based on similar keys to produce lightly harmonized metadata.
> We do this for convenience and to aid in searches.

Reviewer 2: Maybe one sentence?
```suggestion
To aid in searches and for general convenience we combine certain fields based on similar keys to produce lightly harmonized metadata.
```

Author: ```suggestion
To aid in searches and for general convenience, we combine certain fields based on similar keys to produce lightly harmonized metadata.
```

_See the documentation sidebar for the current version of refine.bio._
_The `refinebio_processor_version` field in the downloaded metadata file captures the refine.bio version when a sample was processed._

Scientists who upload results don't always use the same names for related values.
This makes it challenging to search across datasets.
We have implemented some processes to smooth out some of these issues.

![harmonized-metadata](https://user-images.githubusercontent.com/15315514/44549202-5eefc800-a6ee-11e8-8a7b-57826f0153f2.png)

To aid in searches and for general convenience, we combine certain fields based on similar keys to produce lightly harmonized metadata.
For example, `treatment`, `treatment group`, `treatment protocol`, `drug treatment`, and `clinical treatment` fields get collapsed down to `treatment`.
The fields that we currently collapse to includes `specimen part`, `genetic information`, `disease`, `disease stage`, `treatment`, `race`, `subject`, `compound`, `cell_line`, and `time`.

See the table below for the mappings between the keys from source data and the harmonized keys.

> We check a variety of fields from source repositories for the source key values, e.g., the source keys `age`, `characteristic [age]`, and `characteristic_age` would all map to the harmonized key `age`.

Author: How bout:
```suggestion
In addition to the source data keys explicitly listed in the table, we check for variants in the metadata from the source repositories, e.g., the source keys `age`, `characteristic [age]`, and `characteristic_age` would all map to the harmonized key `age`.
```

> We check a variety of fields from source repositories for the source key values, e.g., the source keys `age`, `characteristic [age]`, and `characteristic_age` would all map to the harmonized key `age`.

Reviewer 2: Seems good.

| `compound` | `compound`, `compound1`, `compound2`, `compound name`, `drug`, `drugs`, `immunosuppressive drugs` |
| `time` | `initial time point`, `start time`, `stop time`, `time point`, `sampling time point`, `sampling time`, `time post infection` |
| `age` | `age`, `patient age`, `age of patient`, `age (years)`, `age at diagnosis`, `age at diagnosis years`, `characteristic [age]`, `characteristics [age]` |
| `cell_line` | `cell line` |

When multiple source keys that map to the same harmonized key are present in metadata from sources, we concatenate the values, separated by `;`.

> For example, a sample with `cell type: B cell` and `tissue: kidney` would become `specimen_part: B cell;kidney` when harmonized.

Reviewer 2: Minor question: do we do any sorting on these or is it haphazard (order of the source keys). I might suggest we sort alphabetically, but I can see arguments against, and it isn't particularly important. 

Either way, we should probably make a note about the order of the concatenated values, if only to say it is not meaningful.

Author: Tagging @davidsmejia to weigh in on how concatenation is implemented. I will update the text to reflect the answer.

Reviewer 3: We don't sort the values that get harmonized but the order of the keys that we check is consistent. That order is strongly correlated to the order of what is in the table but that looks a bit dated. I can provide an exhaustive list of the checked source fields for each harmonized value if we want to better represent that here.

Reviewer 3: When I say "dated" I actually mean that the list of source values for example `time` should also contain`time`. This is not true for every harmonized key. Also that the way we create variants means that we should either list all possible way `characteristic` is checked or omit it and say that we check a variety of fields that would have that substring in them. ie. `characteristic [ <attribute> ]` and `characteristic_<attribute>` etc

Author: Would it be difficult to sort by alphabetical order then? Because we could say that the values are in the order that they were checked and update the table here, but if we could say we sorted it alphabetically that's much easier to explain (and maintain re: the table ordering).

> Also that the way we create variants means that we should either list all possible way `characteristic` is checked or omit it and say that we check a variety of fields that would have that substring in them. ie. `characteristic [ <attribute> ]` and `characteristic_<attribute>` etc

I don't think listing all possible ways is going to be easy to maintain, so let's go the other way.

Author: Can you both take a look at https://github.com/AlexsLemonade/refinebio-docs/pull/172/commits/c641186bceef0cf318551fcc8a5ca90da544f31a, @davidsmejia and @jashapiro?

Reviewer 3: > Would it be difficult to sort by alphabetical order then? Because we could say that the values are in the order that they were checked and update the table here, but if we could say we sorted it alphabetically that's much easier to explain (and maintain re: the table ordering).

Yeah that makes sense to me. I will update the implementation to sort the harmonized values. They will be in alphanumeric ascending order. So 0-9 and a-z.
